File: 1889-check-if-number-is-a-sum-of-powers-of-three/check-if-number-is-a-sum-of-powers-of-three.go
func checkPowersOfThree(n int) bool {
    threePow := 4782969 // 3^14 = 4782969

    for threePow > 0 {
        if threePow <= n {
            n -= threePow
        }
        threePow /= 3
    }
    return n == 0
    
}

File: 1889-check-if-number-is-a-sum-of-powers-of-three/check-if-number-is-a-sum-of-powers-of-three.java
class Solution {
    public boolean checkPowersOfThree(int n) {
        int threePow = 4782969; // 3^14 = 4782969
        while( threePow > 0) {
            if( threePow <= n ) {
                n -=threePow;
            }

            threePow /=3;
        }
        return n == 0;  
    }
}